### PR TITLE
Fix Issue 345

### PR DESCRIPTION
**Pull request by cewing**
This PR restores the behavior of a payment timeout resulting in a status of "Payment Not Provided" for work party registrations in Salesforce.  It completes the work to fix Issue #345

### CODE REVIEW

File: src/vms/register.py [hunk 1]
```diff
@@ -561,6 +561,7 @@ def handle_cancel(context, request):
     except ValueError:
         req = {}
     regId = req.get('cancelReg')
+    reg_status = req.get('reg_status', 'Cancelled')
```
Comment (davisagli) [inline src/vms/register.py]: Please validate that the status in the request is either 'Payment Not Provided' or 'Cancelled'.

Comment (cewing) [inline src/vms/register.py]: Okay, If it isn't either, should the request be rejected, or should we override it with a default value?

Comment (cewing) [inline src/vms/register.py]: never mind.  silly question.  I've fixed it by raising an error for a bad status.

Comment (davisagli) [inline src/vms/register.py]: Probably raise an HTTPUnprocessableEntity like we do when validation fails. This isn't expected, I just want to make sure this isn't a backdoor to set other statuses we haven't anticipated it being used for.

Comment (cewing) [inline src/vms/register.py]: fixed.  waiting for tests to pass, then I'll merge

**Pull request merged**